task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -213,40 +213,48 @@ def xcom_pull(
             run_id = self.run_id
 
         if task_ids is None:
+            # default to the current task if not provided
             task_ids = self.task_id
         elif not isinstance(task_ids, str) and isinstance(task_ids, Iterable):
-            # TODO: Handle multiple task_ids or remove support
-            raise NotImplementedError("Multiple task_ids are not supported yet")
-
+            # Retain the ordering as per legacy
+            task_ids = list(task_ids)
```

Comment:
Why do we need to do this at all? Couldn't we just iterate over what is given to us?

Reply:
Right yes. Not needed. I converted it to a set earlier. That would remove the ordering, Ill change it

Reply:
Yeah! we should just keep is as-is

Reply:
Made a change related to this